meta/statement_executor.go

```diff
@@ -26,7 +26,7 @@ type StatementExecutor struct {
 		CreateUser(name, password string, admin bool) (*UserInfo, error)
 		UpdateUser(name, password string) error
 		DropUser(name string) error
-		SetPrivilege(username, database string, p influxql.Privilege) error
+		SetPrivilege(username, database string, p influxql.Privilege, admin bool) error
```

Comment:
You mean that the admin flag should be checked if no database is specified? What if a database is specified and the user has admin privileges?